### PR TITLE
Make file input field not required when editing

### PR DESCRIPTION
**Pull request by SableWalnut**
Change input template for file input to ensure the required attribute is not present.

Fix for #17.

### CODE REVIEW

**Comment by timheap** [comment]
This should be done by changing the field definition in the form, rather than overwriting the render template.

Additionally, some tests should be added to check that the field still works when submitting no file, to ensure that the file field is not cleared for example.

**Comment by SableWalnut** [comment]
@timheap Updated it to not be changed at a template level. There seems to already be tests that cover this. 

```
    def test_edit(self):
        response = self.post({
            'title': "Edited",
        })

        # Should redirect back to index
        self.assertRedirects(response, reverse('wagtailvideos:index'))

        # Check that the video was edited
        video = Video.objects.get(id=self.video.id)
        self.assertEqual(video.title, "Edited")
```
Does not update the video. :)

**Comment by timheap** [comment]
Assuming you're talking about the test https://github.com/takeflight/wagtailvideos/blob/04cbb21/tests/test_admin_views.py#L220-L230, that test does not check that the `video.file` does not change. The `video.file` field could be set to an empty value as part of the save and that test would still pass. A line could be added to the end of that test that checks that the `video.file` field has not changed.

**Comment by SableWalnut** [comment]
@timheap Ah, I get what you mean. Added it! :)

File: wagtailvideos/views/videos.py [hunk 1]
```diff
@@ -76,7 +75,7 @@ def index(request):
 @permission_checker.require('change')
 def edit(request, video_id):
     VideoForm = get_video_form(Video)
-
+    VideoForm.use_required_attribute = False
```
Comment (timheap) [inline wagtailvideos/views/videos.py]: This will also have the unintended side effect of making the `title` field not required, which is not desired.

Comment (SableWalnut) [inline wagtailvideos/views/videos.py]: It'll still raise a validation error if the title field is not filled out, it just won't stop it at a browser level

**Comment by timheap** [comment]
Something like this might work, in `wagtailvideos/forms.py`:

```python
class BaseVideoForm(BaseCollectionMemberForm):
    permission_policy = video_permission_policy

    def __init__(self, *args, **kwargs):
        super(BaseVideoForm, self).__init__(*args, **kwargs)
        # A file is only required if there is not already a file, such as when
        # editing an existing video.
        self.fields['file'].required = 'file' not in self.initial_data
```

It will change whether the field is required depending upon whether it already has a value.

**Comment by SableWalnut** [comment]
@timheap That worked excellently, thanks! Should be all ok now! :)

**Comment by timheap** [comment]
The tests are failing because the code I supplied was not correct - I probably should have tested it but hey that is what a test suite is for! The multiple upload form does not have a `file` field, so some cases were raising a `KeyError`. Additionally, a new instance would still provide an empty value for the `file` key, so the `'file' in self.initial` was not checking the right thing. Try this instead:

```python
class BaseVideoForm(BaseCollectionMemberForm):
    permission_policy = video_permission_policy

    def __init__(self, *args, **kwargs):
        super(BaseVideoForm, self).__init__(*args, **kwargs)
        # A file is only required if there is not already a file, such as when
        # editing an existing video.  The file field is not used on the
        # multiple-upload forms, so may not be present
        if 'file' in self.fields:
            self.fields['file'].required = 'file' not in self.initial or not self.initial['file']
```

Additionally, you've made changes to unrelated files now, mostly whitespace changes in the import list. Check the diff to see what has happened.

**Comment by SableWalnut** [comment]
@timheap Fixed that up. The whitespace and other unrelated changes were flake8 and isort fixes. Without those the test will not pass.

**Pull request merged**